### ISSUE TITLE
dates() method duplicates dates in period with daylight time switch

### ISSUE THREAD

**Issue by max-mykhailenko** [Teamweek/instadate#13]

**Comment by herkyl** [comment]
Could you please write an example of how to replicate it

**Comment by max-mykhailenko** [comment]
Problem was in date object.
`new Date('2016/10/01 00:00:00')` — works wrong (switch timezone — correct)
`new Date(2016, 10, 1, 0, 0, 0)` — works correct (doesn't switch timezone — wrong?)

Can you advise some overall rules for working with `Date` object?

<img width="957" alt="screen shot 2016-10-12 at 9 56 40 am" src="https://cloud.githubusercontent.com/assets/1213198/19300301/385962d6-9062-11e6-8426-941614cb0be5.png">
https://jsbin.com/pojeveratu/edit?js,console

**Comment by herkyl** [comment]
Date timezone is in the local timezone of the user. I recommend not using 00:00:00 times since you can lose a day due to daylight saving time. You can use `instadate.noon` to prevent that.

**Comment by max-mykhailenko** [comment]
So I should to parse time string like `2016/10/01 14:33:13` and set every parameter as argument of `new Date()` for correct work? How I can create `Date` from string without parsing and without incorrect timezone behaviour?

**Comment by herkyl** [comment]
From where are you getting the dates? You can use the [ISO 8601](https://en.wikipedia.org/wiki/ISO_8601) format to safely set them. Most APIs should be returning that. There is no way around the default timezone when using `Date` (or `instadate` since it depends heavily on `Date`). If timezones are important I recommend checking out [Moment.js](http://momentjs.com/) for that part.

If you are generating dates on client-side JS code I recommend creating a base `instadate.noon` date and using instadate to manipulate that (add days, etc).

Sorry for the late replies btw, I have so many GH notifications that I rarely read them..

**Comment by max-mykhailenko** [comment]
Oops, I have mistake in example
https://jsbin.com/nudalik/1/edit?js,console
With `new Date(2016, 9, 1, 0, 0, 0),` I have same mistake in dates array

**Comment by max-mykhailenko** [comment]
I don't want use timezones, but also I don't want any side effects from them

**Comment by herkyl** [comment]
What are you trying to do exactly? If you don't care about timezones instadate should work. Why are you trying to use the format `2016/10/01 00:00:00`?

**Comment by max-mykhailenko** [comment]
After using `dates()`function I expect to get array of unique dates, but I got duplicates :(

**Comment by herkyl** [comment]
This is because of Daylight savings clock turning on October 30 (http://www.webexhibits.org/daylightsaving/b.html).

You can fix this by using the noon function from instadate:
```
var datess = instadate.dates(
  instadate.noon(new Date('2016/10/01 00:00:00')),
  instadate.noon(new Date('2016/11/01 00:00:00'))
);
```

If you are using the 00 hour of the day then daylight saving can shift it to the previous day.

**Comment by max-mykhailenko** [comment]
But setting to noon ruins my time!

**Comment by herkyl** [comment]
You have to find a way around it, as it is a limitation of the JS Date.

**Comment by max-mykhailenko** [comment]
But you can avoid this situation, just don't use 24 hours constant as day length. Instead you can use a function like

```js
function addDays(date, days) {
    var result = new Date(date);
    result.setDate(result.getDate() + days);
    return result;
}
```

**Comment by herkyl** [comment]
I added tests for the edge case as well as changed all of the adding functions in https://github.com/Teamweek/instadate/commit/ce0352aab2fc7de5a3118cec3468dbdf6e02b907 and published it on npm as 0.3.1.

Now I wonder if `differenceInX` functions also have that issue. Need to check those later. Does this fix your issue.

**Comment by max-mykhailenko** [comment]
Thanks! It's great, I'll test it on this week
